### PR TITLE
Enum fixes

### PR DESCRIPTION
**Pull request by agrover**
These save some repeating of enum values by using some crate-provided macros, and also
fix some overly verbose text about RAID1 in its error string.

### CODE REVIEW

File: src/dbus_consts.rs [hunk 1]
```diff
@@ -109,9 +109,7 @@ impl HasCodes for StratisRaidType {
         match *self {
             StratisRaidType::STRATIS_RAID_TYPE_UNKNOWN => "Ok",
             StratisRaidType::STRATIS_RAID_TYPE_SINGLE => "Single",
-            StratisRaidType::STRATIS_RAID_TYPE_RAID1 => {
-                "Mirror between two disks. For 4 disks or more, they are RAID10"
-            }
+            StratisRaidType::STRATIS_RAID_TYPE_RAID1 => "Mirror between two disks",
```
Comment (mulkieran) [inline src/dbus_consts.rs]: Less is more! How about just "Mirrored". Also, while you're here, shouldn't string for RAID_TYPE_UNKNOWN be something like "Unknown"?

File: src/dbus_api.rs [hunk 2]
```diff
@@ -129,10 +127,11 @@ fn get_cache_object_path(m: &Message) -> MethodResult {
     Ok(vec![m.method_return().append3("/dbus/cache/path", 0, "Ok")])
 }
 
-fn get_list_items<T: HasCodes + Display>(m: &Message, items: Iter<T>) -> MethodResult {
-
-    let msg_vec = items
-        .map(|item| {
+fn get_list_items<I, T>(m: &Message, iter: I) -> MethodResult
```
Comment (mulkieran) [inline src/dbus_api.rs]: Can you put T before I?

File: src/dbus_consts.rs [hunk 3]
```diff
@@ -101,48 +82,21 @@ impl HasCodes for StratisErrorEnum {
...
-                                              StratisRaidType::STRATIS_RAID_TYPE_RAID5,
-                                              StratisRaidType::STRATIS_RAID_TYPE_RAID6];
-        TYPES.into_iter()
+custom_derive! {
+    #[derive(Copy, Clone, EnumDisplay,
+             IterVariants(StratisDBusRaidTypeVariants),
+             IterVariantNames(StratisDBusRaidTypeVariantNames))]
+    #[allow(non_camel_case_types)]
+    pub enum StratisRaidType {
+        STRATIS_RAID_TYPE_UNKNOWN,
```
Comment (mulkieran) [inline src/dbus_consts.rs]: Make comments go away, for they are redundant?

File: src/dbus_api.rs (hunk 2 above)
Comment (mulkieran) [inline src/dbus_api.rs]: Because I depends on it.

**Comment by agrover** [comment]
updated, please take a look @mulkieran @trgill

**Pull request merged**